### PR TITLE
feat(avatar): add styles for default (medium) size in ionic theme

### PR DESCRIPTION
**Pull request by brandyscarney**
Issue number: internal

---------

## What is the current behavior?
Avatar does not have any styles in the ionic theme.

## What is the new behavior?
- Adds background, border and font styles for avatar
- Adds the styles for the medium size and defaults the size to medium
- Adds e2e test for avatar sizes

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

[Preview](https://ionic-framework-git-rou-10735-ionic1.vercel.app/src/components/avatar/test/size?ionic:theme=ionic)

### CODE REVIEW

File: core/src/components/avatar/avatar.ionic.scss [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  background: globals.$ionic-color-neutral-100;
+  color: globals.$ionic-color-neutral-800;
+
+  font-weight: globals.$ionic-font-weight-regular;
+
+  line-height: globals.$ionic-line-height-600;
+}
+
+:host(:not(.avatar-image)) {
+  border: 1px solid globals.$ionic-color-neutral-800;
```
Comment (thetaPC) [inline core/src/components/avatar/avatar.ionic.scss:22]:
```suggestion
  border: globals.$ionic-border-size-025 solid globals.$ionic-color-neutral-800;
```

File: core/src/components/avatar/test/size/avatar.e2e.ts-snapshots/avatar-size-medium-image-ionic-md-ltr-light-Mobile-Chrome-linux.png
Comment (thetaPC) [inline core/src/components/avatar/test/size/avatar.e2e.ts-snapshots/avatar-size-medium-image-ionic-md-ltr-light-Mobile-Chrome-linux.png:1]: Just thinking out loud, not part of this PR: I wonder if we there's a way for Playwright to not had that extra space. It makes it seem that it's part of the background of avatar.

**Review by thetaPC — CHANGES_REQUESTED** [review]
Minor change

File: core/src/components/avatar/test/size/avatar.e2e.ts-snapshots/avatar-size-medium-image-ionic-md-ltr-light-Mobile-Chrome-linux.png
Comment (brandyscarney) [reply core/src/components/avatar/test/size/avatar.e2e.ts-snapshots/avatar-size-medium-image-ionic-md-ltr-light-Mobile-Chrome-linux.png:1]: I don't see any extra space. Maybe it's the GitHub interface?

**Review by BenOsodrac — APPROVED** [review]

**Review by OS-giulianasilva — APPROVED** [review]

**Pull request merged**